astropy/io/fits/src/compressionmodule.c

```diff
@@ -233,159 +235,132 @@ int compress_type_from_string(char* zcmptype) {
 }
 
 
+PyObject *
+get_header_value(PyObject* header, char* key) {
+    PyObject* hdrkey;
+    PyObject* hdrval;
+    hdrkey = PyUnicode_FromString(key);
+    if (hdrkey == NULL) {
+        return NULL;
+    }
+    hdrval = PyObject_GetItem(header, hdrkey);
+    Py_DECREF(hdrkey);
+    PyErr_Clear();
```

Comment:
I haven't looked at the C API in a while so this is more of a question than a comment: why is this the only function where `PyErr_Clear` is being used? Does that imply that there's an error here that we're choosing not to handle?

Reply:
Yes, `PyObject_GetItem` could fail, say with a `KeyError` in case the keyword wasn't present in the Header. In that case `hdrval` is `NULL` but that doesn't matter because I do catch the case where it's `NULL` in the `get_header_*` functions.

Reply:
If the function had some other kind of unhandled exception shouldn't it get propagated back up to user code?

Reply:
You mean anything except `KeyError`?

Reply:
Yes, for example can the `PyUnicode_FromString` above cause an error that we would inadvertently clear here?

Reply:
If `PyUnicode_FromString` raises an Exception the `hdrkey` would be `NULL` and it would exit the function before it comes to the `PyErr_Clear`. Only in case `PyObject_GetItem` fails the exception is cleared. It probably should only clear if there had been a `KeyError`. But the previous implementation also cleared all errors that happened with `GetItem`.